pkg/dataplane/identity.go

```diff
@@ -69,14 +82,122 @@ func (c CredentialsObject) GetCredential(requestedResourceID string) (*azidentit
 
 // Get an AzIdentity credential for the given nested credential object
 // Clients can use the credential to get a token for the user-assigned identity
-func (n NestedCredentialsObject) GetCredential() (*azidentity.ClientCertificateCredential, error) {
+func (n *NestedCredentialsObject) GetCredential() (*azidentity.ClientCertificateCredential, error) {
 	if n.Values.ResourceID != nil {
 		return getClientCertificateCredential(n.Values, n.Cloud)
 	}
 
 	return nil, errResourceIDNotFound
 }
 
+func (n *NestedCredentialsObject) ReloadCredntialsOnChange() error {
+	if err := n.File.checkFileExists(); err != nil {
```

Comment:
Instead of thinking about all the different ways in which the file might be invalid - not existing is just one of them - let's try loading it once on startup and return any errors which we see. This keeps us from having to write code that might or might not be a sufficient validation check.